pkg/infrastructure/azure/azure.go

```diff
@@ -471,6 +473,7 @@ func (p *Provider) InfraReady(ctx context.Context, in clusterapi.InfraReadyInput
 		lbBaps = loadBalancer.Properties.BackendAddressPools
 		extLBFQDN = *publicIP.Properties.DNSSettings.Fqdn
 		pubIPAddress = *publicIP.Properties.IPAddress
+		p.publicLBIP = pubIPAddress
```

Comment:
nit: 
```suggestion
		p.publicLBIP = *publicIP.Properties.IPAddress
```
Then rename existing references of `pubIPAddress` -> p.publicLBIP